src/store/blocks.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable no-param-reassign */
+/*
+   This file is part of Astarte.
+
+   Copyright 2021 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import AstarteClient, { AstarteBlock, AstarteCustomBlock } from 'astarte-client';
+import _ from 'lodash';
+
+import type { StoreState } from './index';
+
+interface BlocksState {
+  blocks: AstarteBlock[] | null;
+  status: 'empty' | 'loading' | 'ok' | 'err';
+  error: string | null;
+  errorByBlock: { [blockName: string]: string };
+  blocksBeingRegistered: string[];
+  blocksBeingUpdated: string[];
+  blocksBeingDeleted: string[];
+}
+
+const initialState: BlocksState = {
+  blocks: null,
+  status: 'empty',
```

Comment:
'empty` is an invalid state that represents that we did not retrieve the blocks from the Astarte API.
While in this state, no action should be allowed except loading blocks.
If that initial loading fails, you end up in the error state with a null value for blocks.

What about starting in a "valid" initial state? Like `ok` and empty list of blocks?